File: cpp/UVA10189.cpp
//https://zerojudge.tw/ShowProblem?problemid=e605
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
    // 優化輸入輸出速度 (競賽必備)
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n, m, cases = 1;
    
    // 當讀入 n, m 且兩者不全為 0 時繼續執行
    while (cin >> n >> m && (n || m)) {
        // 處理格式：除了第一筆測資外，前面的輸出都要多換一行
        if (cases > 1) cout << '\n';
        cout << "Field #" << cases++ << ":\n";

        // 讀取地圖
        vector<string> g(n);
        for (int i = 0; i < n; i++) {
            cin >> g[i];
        }

        // 掃描每個格子
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (g[i][j] == '*') {
                    cout << '*'; // 是地雷直接輸出
                } else {
                    int count = 0;
                    // 檢查九宮格 (包含自己也沒關係，因為自己不是 '*')
                    for (int dy = -1; dy <= 1; dy++) {
                        for (int dx = -1; dx <= 1; dx++) {
                            int ny = i + dy;
                            int nx = j + dx;
                            // 確保邊界不超範圍，且該位置為地雷
                            if (ny >= 0 && ny < n && nx >= 0 && nx < m && g[ny][nx] == '*') {
                                count++;
                            }
                        }
                    }
                    cout << count;
                }
            }
            cout << '\n'; // 每一列結束後換行
        }
    }

    return 0;
}

File: java/UVA10189.java
//https://zerojudge.tw/ShowProblem?problemid=e605
import java.util.Scanner;

class UVA10189{
	public static void main(String args[]){
		Scanner sc=new Scanner(System.in);
		
		int cases=sc.nextInt();
		for(int i=0;i<cases;i++){
			String st1=sc.next();
			String st2=sc.next();
			
			//binary to integer
			int st1N=0;
			int st2N=0;
			for(int j=st1.length()-1;j>=0;j--){
				if(st1.charAt(j)!='0') st1N=st1N+(int)Math.pow(2,st1.length()-1-j);
			}
			for(int j=st2.length()-1;j>=0;j--){
				if(st2.charAt(j)!='0') st2N=st2N+(int)Math.pow(2,st2.length()-1-j);
			}
			
			//gcd
			while(st2N!=0){
				int temp=st2N;
				st2N=st1N%st2N;
				st1N=temp;
			}
			//output
			if(st2N==0 && st1N!=1)System.out.println("Pair #"+(i+1)+": All you need is love!");
			else System.out.println("Pair #"+(i+1)+": Love is not all you need!");
		}
	}
}
